mac_pw_pool/LaunchInstances.sh

```diff
@@ -0,0 +1,283 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# Script intended to be executed by humans (and eventually automation) to
+# ensure instances are launched from the current template version, on all
+# available Cirrus-CI Persistent Worker M1 Mac dedicated hosts.  These
+# dedicated host (slots) are selected at runtime based their possessing any
+# value for the tag `PWPoolReady`.  The script assumes:
+#
+# * The `aws` CLI tool is installed on $PATH.
+# * Appropriate `~/.aws/credentials` credentials are setup.
+# * The us-east-1 region is selected in `~/.aws/config`.
+#
+# N/B: Dedicated Host names and instance names are assumed to be identical,
+# only the IDs differ.
+
+# Name of launch template. Current/default version will be used.
+# https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#LaunchTemplates:
+TEMPLATE_NAME=CirrusMacM1PWinstance
+
+# Item format: "<Dedicated Host / Instance Name> <Dedicated Host ID>"
+# https://us-east-1.console.aws.amazon.com/ec2/home?region=us-east-1#Hosts:
+declare -a NAME2HOST
+
+# Path to file recording one line per instance, with it's name, instance id, start
+# date/time separated by a space.  Exceptional conditions are recorded as comments
+# with the name and details.  File is refreshed/overwritten each time script runs
+# without any fatal/uncaught command-errors.  Intended for reference by humans
+# and/or other tooling.
+PWSTATE="$(dirname ${BASH_SOURCE[0]})/pw_status.txt"
+
+# AWS CLI command and general args
+AWS="aws --no-paginate --output=json --color=off --no-cli-pager --no-cli-auto-prompt"
+
+SCRIPT_FILENAME=$(basename "${BASH_SOURCE[0]}")
+
+TEMPDIR=$(mktemp -d -p '' "${SCRIPT_FILENAME}_XXXXX.tmp")
+
+# Set non-zero to enable debugging / prevent removal of temp. dir.
+L_DEBUG="${L_DEBUG:0}"
+
+dbg() {
+  if ((L_DEBUG)); then
+      echo "${1:-No debug message provided}" | awk -e '{print "DEBUG: "$0}' > /dev/stderr
```

Comment:
Two notes (hopefully they help):

- The LaunchInstances.sh and SetupInstance.sh are intended to run on my laptop for now, not on the mac.  Only setup.sh and service_pool.sh are targeted to run on the macs.
- The awk is there because there are/were sometimes multi-line calls to dbg/warn/die.  The awk simply makes it clear all those lines are for debugging, warning, or death.  As opposed to something else like a script syntax error.